apps/src/craft/code-connection/craft.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import {singleton as studioApp} from '../../StudioApp';
+import codegen from '../../codegen';
+import {Provider} from 'react-redux';
+import AppView from '../../templates/AppView';
+import CraftVisualizationColumn from './CraftVisualizationColumn';
+import {getStore} from '../../redux';
+
+const MEDIA_URL = '/blockly/media/craft/';
+
+/**
+ * Create a namespace for the application.
+ */
+const Craft = module.exports;
+
+window.Craft = Craft;
+window.Blockly = Blockly;
+
+const COMMON_UI_ASSETS = [
+  MEDIA_URL + "Sliced_Parts/MC_Loading_Spinner.gif",
+  MEDIA_URL + "Sliced_Parts/Frame_Large_Plus_Logo.png",
+  MEDIA_URL + "Sliced_Parts/Pop_Up_Slice.png",
+  MEDIA_URL + "Sliced_Parts/X_Button.png",
+  MEDIA_URL + "Sliced_Parts/Button_Grey_Slice.png",
+  MEDIA_URL + "Sliced_Parts/MC_Button_Pressed.png",
+  MEDIA_URL + "Sliced_Parts/Run_Button_Up_Slice.png",
+  MEDIA_URL + "Sliced_Parts/Run_Button_Down_Slice.png",
+  MEDIA_URL + "Sliced_Parts/MC_Run_Arrow_Icon_Smaller.png",
+  MEDIA_URL + "Sliced_Parts/MC_Up_Arrow_Icon.png",
+  MEDIA_URL + "Sliced_Parts/MC_Down_Arrow_Icon.png",
+  MEDIA_URL + "Sliced_Parts/Reset_Button_Up_Slice.png",
+  MEDIA_URL + "Sliced_Parts/MC_Reset_Arrow_Icon.png",
+  MEDIA_URL + "Sliced_Parts/Reset_Button_Down_Slice.png",
+  MEDIA_URL + "Sliced_Parts/Callout_Tail.png",
+];
```

Comment:
These are all probably small enough to be included into the js directly as a data url so you wouldn't have to "preload" them. Then you can just import them directly using webpack's url loader to generate the data url automatically at build time.